src/main/java/nl/tudelft/contextproject/model/level/RandomLevelFactory.java

```diff
@@ -1,108 +1,340 @@
+	public Level generateSeeded(long seed) {
+		createRNG(seed);
+
+		List<GeneratorRoom> generated = create();
+		TileType[][] carved = carveRooms(generated);
+		carved = carveCorridors(carved, generated);
+
+		MazeTile[][] mazeTiles = new MazeTile[MAX_WIDTH][MAX_HEIGHT];
+
+		for (int x = 0; x < carved.length; x++) {
+			for (int y = 0; y < carved[0].length; y++) {
+				if (carved[x][y] != null) {
+					mazeTiles[x][y] = new MazeTile(x, y, carved[x][y]);
 				}
 			}
 		}
+
+		ArrayList<Light> lights = new ArrayList<>(1);
+		PointLight p = new PointLight();
+		p.setPosition(new Vector3f(1, 1, 4));
+		p.setColor(ColorRGBA.randomColor());
+		p.setRadius(20);
+		lights.add(p);
+		return new Level(mazeTiles, lights);
+	}
+
+	/**
+	 * Create the random number generator.
+	 *
+	 * @param seed
+	 * 			the seed to use for the random number generator.
+     */
+	protected void createRNG(long seed) {
+		rand = new Random(seed);
+	}
+
+	/**
+	 * Return a random number between two values.
+	 * @param min
+	 *          The minimum value the random number can be.
+	 * @param max
+	 *          The maximum value the random number can be.
+	 * @return
+	 *          The random number.
+	 */
+	protected int getRandom(int min, int max) {
+		return (rand.nextInt((max - min)) + min);
+	}
+
+	/**
+	 * Method to load all rooms from file and return the sizes.
+	 *
+	 * <p>The URISyntaxException occurs if somehow the URL generated by {@link Class#getResource(String)}
+	 * is invalid. Since this does not occur, this exception can safely be ignored.
+	 *
+	 * @return
+	 *          An ArrayList of sizes.
+	 */
+	@SneakyThrows(URISyntaxException.class)
+	protected List<Size> loadRooms() {
+		File[] files = new File(RandomLevelFactory.class.getResource("/rooms").toURI()).listFiles();
+
+		/**
+		 * If the folder is not a folder, we throw an IllegalStateException exception.
+		 */
+		if (files == null) {
+			throw new IllegalStateException("There are no rooms.");
+		}
+
+		List<Size> sizes = new ArrayList<>();
+		for (File f : files) {
+			Matcher m = PATTERN.matcher(f.getName());
+			if (m.matches()) {
+				int width = Integer.parseInt(m.group("width"));
+				int height = Integer.parseInt(m.group("height"));
+				sizes.add(new Size(width, height));
+			}
+		}
+		return sizes;
 	}
 
-	public int getNeighbourCount(int x, int y) {
-		int result = 0;
-		for (int movX = x - 1; movX <= x + 1; movX++) {
-			for (int movY = y - 1; movY <= y + 1; movY++) {
-				if (movY >= 0 && movY < height && movX >= 0 && movX < width) {
-					if (movY != y && movX != x) {
-						result += preCreate[movX][movY];
+	/**
+	 * Method to create a list of {@link GeneratorRoom}.
+	 *
+	 * @return
+	 * 			the created generatorrooms
+     */
+	protected List<GeneratorRoom> create() {
+		List<GeneratorRoom> rooms = new ArrayList<>();
+		List<Size> sizes = loadRooms();
+		if (!allowDuplicates && sizes.size() < amount) {
+			throw new IllegalArgumentException("You are requesting more rooms than there are.");
```

Comment:
Maybe add sizes.size() to this message to give the user a maximum value to choose from.